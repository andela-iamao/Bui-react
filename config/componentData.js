module.exports = [{"name":"Button","description":"Button Component","props":{"children":{"type":{"name":"node"},"required":true,"description":"Child elements for the button component"},"onClick":{"type":{"name":"func"},"required":false,"description":""},"type":{"type":{"name":"enum","value":[{"value":"'primary'","computed":false},{"value":"'secondary'","computed":false},{"value":"'danger'","computed":false}]},"required":false,"description":""}},"code":"import React from 'react';\nimport PropTypes from 'prop-types';\n\n\nconst colorScheme = {\n    primary: '#236167',\n    secondary: '#000000',\n    danger: '#AA3939'\n}\n\n/**\n * Button Component\n */\nfunction Button({ children, ...props}) {\n    return (\n        <button {...props} style={style(props.type)}>\n            {children}\n        </button>\n    );\n}\n\nconst style = (type) => ({\n    color: '#FFFFFF',\n    backgroundColor: colorScheme[type],\n    padding: 10,\n    width: 120,\n    border: 'none'\n});\n\nButton.propTypes = {\n    /**\n     * Child elements for the button component\n     */\n    children: PropTypes.node.isRequired,\n    onClick: PropTypes.func,\n    type: PropTypes.oneOf(['primary', 'secondary', 'danger'])\n}\n\nexport default Button;\n","examples":[{"name":"DangerButton","description":"Danger Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Danger Button component\n */\nexport default () => {\n    return (\n        <Button type=\"danger\">Danger Button!</Button>\n    );\n}\n"},{"name":"PrimaryButton","description":"Primary Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Primary Button component\n */\nexport default () => {\n    return (\n        <Button type=\"primary\">Primary Button!</Button>\n    );\n}\n"},{"name":"SecondaryButton","description":"Secondary Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Secondary Button component\n */\nexport default () => {\n    return (\n        <Button type=\"secondary\">Secondary Button!</Button>\n    );\n}\n"}]}]