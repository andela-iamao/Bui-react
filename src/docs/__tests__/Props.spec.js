import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Props from '../Props';
import componentData from '../../../config/jest/componentDataMock';


describe('DOCS: Props', () => {

    describe('Snapshot Test', () => {
        test('renders component with props', () => {
            const data = { ...componentData[0]};
            data.description = null;
            const tree = renderer.create(
                <Props props={componentData[0].props} />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
        test('renders component without props', () => {
            const tree = renderer.create(
                <Props props={{}} />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        })
    });

    describe('Unit Test', () => {
        test('renders all props data', () => {
            const data = {
                ...componentData[0].props,
                children: {
                  ...componentData[0].props.children,
                  defaultValue: '<div />'
                }
            };
            const wrapper = shallow(<Props props={data} />);
            expect(wrapper.find('table')).toHaveLength(1);
            expect(wrapper.find('tbody > tr')).toHaveLength(Object.keys(componentData[0].props).length);
        });
        test('renders empty body rows when there are no props', () => {
            const wrapper = shallow(<Props props={{}} />);
            expect(wrapper.find('table')).toHaveLength(1);
            expect(wrapper.find('tbody > tr')).toHaveLength(0);
        });
    });
});
