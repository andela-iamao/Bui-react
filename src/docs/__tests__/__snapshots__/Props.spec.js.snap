// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOCS: Props Snapshot Test renders component with props 1`] = `
<div
  id="bui-react-docs-component-props-container"
>
  <table
    className="props"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Type
        </th>
        <th>
          Default
        </th>
        <th>
          Required
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          children
        </td>
        <td>
          Child elements for the button component
        </td>
        <td>
          node
        </td>
        <td />
        <td>
          Nope
        </td>
      </tr>
      <tr>
        <td>
          onClick
        </td>
        <td>
          
        </td>
        <td>
          func
        </td>
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DOCS: Props Snapshot Test renders component without props 1`] = `
<div
  id="bui-react-docs-component-props-container"
>
  <table
    className="props"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Type
        </th>
        <th>
          Default
        </th>
        <th>
          Required
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
